pybossa/importers.py

```diff
@@ -160,13 +153,39 @@ def _get_album_info(self, album_id):
                    'format': 'json',
                    'nojsoncallback': '1'}
         res = requests.get(url, params=payload)
-        return json.loads(res.text)
+        if self._is_valid_response(res):
+            content = json.loads(res.text)['photoset']
+            total_pages = content.get('pages')
+            rest_photos = self._remaining_photos(url, payload, total_pages)
+            content['photo'] += rest_photos
+            return content
+        if type(res.tex) is dict:
+            error_message = json.loads(res.text).get('message')
+        else:
+            error_message = res.text
+        raise BulkImportException(error_message)
+
+    def _is_valid_response(self, response):
+        return (response.status_code == 200
```

Comment:
you mean to change it like this?

    def _get_album_info(self, album_id):
        url = 'https://api.flickr.com/services/rest/'
        payload = {'method': 'flickr.photosets.getPhotos',
                   'api_key': self.api_key,
                   'photoset_id': album_id,
                   'format': 'json',
                   'nojsoncallback': '1'}
        res = requests.get(url, params=payload)
        if self._is_valid_response(res):
            content = json.loads(res.text)['photoset']
            total_pages = content.get('pages')
            rest_photos = self._remaining_photos(url, payload, total_pages)
            content['photo'] += rest_photos
            return content

    def _is_valid_response(self, response):
        if type(response.tex) is dict:
            error_message = json.loads(response.text).get('message')
        else:
            error_message = response.text
        valid = (response.status_code == 200
                    and json.loads(response.text).get('stat') == 'ok')
        if not valid:
            raise BulkImportException(error_message)
        return valid


Yes, this seems to work ok